### PR TITLE
Fix RBAC 

### PR DESCRIPTION
**Pull request by ukclivecox**
* kubebuilder annotation fix for RBAC. The group and resource kind names were wrong.
 * Add `undeploy` goals to Makefile for easy testing
 * Add `IfNotPresent` to manager yaml image pull for easy minikube testing by adding a kustomize overlay

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/34)

### CODE REVIEW

File: config/default/manager_image_patch.yaml [hunk 1]
```diff
@@ -10,3 +10,4 @@ spec:
       # Change the value of image field below to your controller image URL
       - image: kfserving-controller:latest
         name: manager
+        imagePullPolicy: IfNotPresent
```
Comment (yuzisun) [inline config/default/manager_image_patch.yaml]: for real deployments we may want `PullAlways` though

Comment (cliveseldon) [reply config/default/manager_image_patch.yaml]: I have added a development kustomize overlay

**Comment by cliveseldon** [comment]
I have reverted the `dep ensure`.

File: Makefile [hunk 2]
```diff
@@ -25,6 +25,18 @@ deploy: manifests
 	kubectl apply -f config/crds
 	kustomize build config/default | kubectl apply -f -
 
+deploy-dev: manifests
+	kubectl apply -f config/crds
+	kustomize build config/overlays/development | kubectl apply -f -
+
+undeploy: manifests
```
Comment (ellis-bigelow) [inline Makefile]: Should we call this clean?

File: config/overlays/development/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+bases:
```
Comment (ellis-bigelow) [inline config/overlays/development/kustomization.yaml]: I'm not clear on the "overlays" convention. Can you explain what this means?

File: config/rbac/rbac_role.yaml [hunk 4]
```diff
@@ -25,9 +25,9 @@ rules:
   - update
   - patch
 - apiGroups:
-  - kfserving.kubeflow.org
+  - serving.kubeflow.org
```
Comment (ellis-bigelow) [inline config/rbac/rbac_role.yaml]: +1 on all the changes in this file

File: pkg/reconciler/ksvc/reconciler.go [hunk 5]
```diff
@@ -48,9 +48,9 @@ func (c *ServiceReconciler) Reconcile(ctx context.Context, desiredService *knser
 	err := c.client.Get(context.TODO(), types.NamespacedName{Name: desiredService.Name, Namespace: desiredService.Namespace}, service)
 	if err != nil {
 		if errors.IsNotFound(err) {
-		  log.Info("Creating Knative Serving Service", "namespace", service.Namespace, "name", service.Name)
```
Comment (ellis-bigelow) [inline pkg/reconciler/ksvc/reconciler.go]: @yuzisun needs an autoformatter ;)

File: config/overlays/development/kustomization.yaml (hunk 3 above)
Comment (cliveseldon) [reply config/overlays/development/kustomization.yaml]: I'm following [this](https://github.com/kubernetes-sigs/kustomize#2-create-variants-using-overlays) but happy to defer to more experienced Kustomize users.

File: Makefile (hunk 2 above)
Comment (cliveseldon) [reply Makefile]: Happy to - just wanted to match the auto-generated deploy targets.

File: config/rbac/rbac_role.yaml [hunk 6]
```diff
@@ -37,9 +37,9 @@ rules:
   - patch
   - delete
 - apiGroups:
-  - kfserving.kubeflow.org
+  - serving.kubeflow.org
   resources:
-  - services/status
+  - kfservices/status
   verbs:
```
Comment (animeshsingh) [inline config/rbac/rbac_role.yaml]: So the docs are referring to ' kfserving' resource, and then we flip between kfservices and serving.....would it make sense to be consistent and prepend everything with kf?

File: pkg/reconciler/ksvc/reconciler.go (hunk 5 above)
Comment (animeshsingh) [reply pkg/reconciler/ksvc/reconciler.go]: Should the log be "Creating KF Serving Service Instance" ?

File: config/overlays/development/kustomization.yaml (hunk 3 above)
Comment (ellis-bigelow) [reply config/overlays/development/kustomization.yaml]: Makes complete sense. Thanks!

File: pkg/reconciler/ksvc/reconciler.go (hunk 5 above)
Comment (yuzisun) [reply pkg/reconciler/ksvc/reconciler.go]: oops, might forgot to run `make fmt`, need to force this in CI.

File: Makefile (hunk 2 above)
Comment (rakelkar) [reply Makefile]: is this folder structure working for you guys... I get this sad error: 
Error: rawResources failed to read Resources: Load from path ../rbac/rbac_role.yaml failed: security; file '../rbac/rbac_role.yaml' is not in or below '/home/rakelkar/go/src/github.com/kubeflow/kfserving/config/default' 

You're possibly using an older version of kustomize?
https://github.com/kubernetes-sigs/kustomize/issues/766

Comment (cliveseldon) [reply Makefile]: I think this is an issue with kubebuilder which is fixed in the unreleased master : kubernetes-sigs/kubebuilder/issues/654

File: config/rbac/rbac_role.yaml (hunk 6 above)
Comment (ellis-bigelow) [reply config/rbac/rbac_role.yaml]: kfservice is one of the resources in the group serving.kubeflow.org. Another resource might be kfgraph in serving.kubeflow.org. Make sense?

File: Makefile (hunk 2 above)
Comment (cliveseldon) [reply Makefile]: I think we need to decide to staywith current kubebuilder release v1.0.8 and kustomize 1.x and then move to kustomize 2.x when kubebuilder releases what is in HEAD?

**Comment by ellis-bigelow** [comment]
I'm happy to push this Clive, just remove the Gopkg.lock. I think you missed it :)

File: config/default/manager_image_patch.yaml (hunk 1 above)
Comment (ellis-bigelow) [reply config/default/manager_image_patch.yaml]: Don't we want PullAlways for the devloop too?

Comment (cliveseldon) [reply config/default/manager_image_patch.yaml]: For local minikube testing I build the image in minikube so don't want to pull from anywhere. For external development yes I suppose you do. Maybe if we have a preferred development guide we could sync on what is best practise. Happy to deploy internal docker repo to cluster as well in which case always pulling makes sense. Shall we merge this PR and then update with advice from @yuzisun and yourself?

**Comment by cliveseldon** [comment]
Ping @ellis-bigelow  - ready to approve?

**Pull request merged**